### PR TITLE
Added error handling in ValidateAdmissionPolicy

### LINKED ISSUE #114726: Found some FIXME in validatingadmissionpolicy testing file
https://github.com/kubernetes/kubernetes/blob/7eb9a75bf914428e7579d2497722f71be3ef1812/staging/src/k8s.io/apiserver/pkg/admission/plugin/validatingadmissionpolicy/admission_test.go#L255

https://github.com/kubernetes/kubernetes/blob/7eb9a75bf914428e7579d2497722f71be3ef1812/staging/src/k8s.io/apiserver/pkg/admission/plugin/validatingadmissionpolicy/admission_test.go#L260

### PR DESCRIPTION
**Pull request by dipankardas011**
Signed-off-by: Dipankar Das <dipankardas0115@gmail.com>

#### What type of PR is this?

/kind bug
/kind cleanup

#### What this PR does / why we need it:
there is no error handling in the mentioned lines

#### Which issue(s) this PR fixes:

Fixes #114726

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by aojea** [comment]
/ok-to-test
/assign @DangerOnTheRanger 

he volunteered to follow up on this https://github.com/kubernetes/kubernetes/issues/114726#issuecomment-1374185371

File: staging/src/k8s.io/apiserver/pkg/admission/plugin/validatingadmissionpolicy/admission_test.go [hunk 1]
```diff
@@ -252,13 +252,11 @@ func setupTestCommon(t *testing.T, compiler ValidatorCompiler, shouldStartInform
...
 	if err != nil {
-		// FIXME: handle error.
-		panic("Unexpected error")
+		t.Fatalf("Unable to add featureGate with Release as Alpha")
 	}
 	err = featureGate.SetFromMap(map[string]bool{string(features.ValidatingAdmissionPolicy): true})
 	if err != nil {
-		// FIXME: handle error.
-		panic("Unexpected error.")
+		t.Fatalf("Unable to Set key ValidateAdmissionPolicy to the Alpha featureGate")
```
[the remark this answers is not in the document]
Comment (dipankardas011) [reply staging/src/k8s.io/apiserver/pkg/admission/plugin/validatingadmissionpolicy/admission_test.go:259]: Will fix that😇

File: staging/src/k8s.io/apiserver/pkg/admission/plugin/validatingadmissionpolicy/admission_test.go [hunk 2]
```diff
@@ -252,13 +252,11 @@ func setupTestCommon(t *testing.T, compiler ValidatorCompiler, shouldStartInform
 		features.ValidatingAdmissionPolicy: {
 			Default: true, PreRelease: featuregate.Alpha}})
 	if err != nil {
-		// FIXME: handle error.
-		panic("Unexpected error")
+		t.Fatalf("Unable to add featureGate with Release as Alpha")
```
[the remark this answers is not in the document]
Comment (dipankardas011) [reply staging/src/k8s.io/apiserver/pkg/admission/plugin/validatingadmissionpolicy/admission_test.go:255]: Will fix that 😇

**Comment by dipankardas011** [comment]
@DangerOnTheRanger is this PR mergeable?

**Pull request merged**